relayer/client/txs.go

```diff
@@ -58,15 +58,9 @@ func (client *servicerClient) timeoutTxs(
 	ctx context.Context,
 	blocksNotifee utils.Observable[types.Block],
 ) {
-	ch := blocksNotifee.Subscribe().Ch()
+	ch := blocksNotifee.Subscribe(ctx).Ch()
 	// TODO: Add a comment
 	for block := range ch {
-		select {
```

Comment:
This is now handled at the subscriber level. We pass `ctx` so it handles it on its own instead of setting up the context termination each time.

It is definitely not related to this feature.